src/earcut.js

```diff
@@ -347,54 +317,59 @@ function splitEarcut(start, triangles, minX, minY, size) {
 }
 
 // link every hole into the outer loop, producing a single-ring polygon without holes
-function eliminateHoles(points, outerNode) {
-    var len = points.length;
-
-    var queue = [];
-    for (var i = 1; i < len; i++) {
-        var list = filterPoints(linkedList(points[i], false));
-        if (list) queue.push(getLeftmost(list));
+function eliminateHoles(data, holeIndices, outerNode, dim) {
+    var queue = [],
+        i, len, start, end, list;
+
+    for (i = 0, len = holeIndices.length; i < len; i++) {
+        start = holeIndices[i] * dim;
+        end = i < len - 1 ? holeIndices[i + 1] * dim : data.length;
+        list = filterPoints(data, linkedList(data, start, end, dim, false));
+        if (list) queue.push(getLeftmost(data, list));
     }
-    queue.sort(compareX);
+
+    queue.sort(function (a, b) {
+        return data[a.i] - data[b.i];
+    });
```

Comment:
I think you need lexicographic order order here, taking the y-coord into account as well in case of the x-coords being equal, something like this:
``` JS
function(a, b) {
    var diff = data[a.i] - data[b.i];
    if(diff !== 0) { return diff; }
    return (data[a.i +1] - data[b.i + 1]);
}
```